runners/java-fn-execution/src/test/java/org/apache/beam/runners/fnexecution/control/DefaultJobBundleFactoryTest.java

```diff
@@ -459,27 +461,28 @@ public void loadBalancesBundles() throws Exception {
       verify(envFactory, Mockito.times(2)).createEnvironment(eq(environment), any());
 
       long tms = System.currentTimeMillis();
-      AtomicBoolean closed = new AtomicBoolean();
-      // close to free up environment for another bundle
-      TimerTask closeBundleTask =
-          new TimerTask() {
-            @Override
-            public void run() {
-              try {
-                b2.close();
-                closed.set(true);
-              } catch (Exception e) {
-                throw new RuntimeException(e);
-              }
-            }
-          };
-      new Timer().schedule(closeBundleTask, 100);
-
+      ScheduledExecutorService executor = Executors.newSingleThreadScheduledExecutor();
+      ScheduledFuture<Optional<Exception>> closingFuture =
+          executor.schedule(
+              () -> {
+                try {
+                  b2.close();
+                  return Optional.empty();
+                } catch (Exception e) {
```

Comment:
nit: You can drop this, `Future` will catch exceptions, wrap them in a `ExecutionException` and throw them out of `get`.

Reply:
Hm could it be that Future only does that for unchecked exceptions? I had tried just making this `executor.schedule(b2::close, ..)` but it produces a compile error:
```
incompatible thrown types Exception in method reference
```

Reply:
Yes only unchecked exceptions. It is odd to me that the Java `AutoCloseable` interface defines it as throwing `Exception`, but it does.